postgresql/src/main/java/org/apache/zeppelin/postgresql/PostgreSqlInterpreter.java

```diff
@@ -264,7 +286,13 @@ public Scheduler getScheduler() {
 
   @Override
   public List<String> completion(String buf, int cursor) {
-    return null;
+
+    List<CharSequence> candidates = new ArrayList<CharSequence>();
+    if (sqlCompleter.complete(buf, cursor, candidates) >= 0) {
+      return Lists.transform(candidates, sequenceToStringTransformer);
+    } else {
+      return NO_COMPLETION;
```

Comment:
I like that you are preallocating it. Could you also avoid allocate `candidates` by moving it into the True clause in line 292?